### ISSUE TITLE
`TypeVar` not resolved when used with nested `Annotated`

### ISSUE THREAD

**Issue by NCPlayz** [microsoft/pyright#3824]
**Describe the bug**
It seems like when using a `TypeVar` with a nested `Annotated` providing a type for the `TypeVar` does not work.

**To Reproduce**
See code.

**Expected behavior**
`JSON`'s value should narrow down to `Dict[str, Any]`

**Screenshots or Code**
```py
from typing import Annotated, Dict, Any, TypeVar

T = TypeVar('T')

Consume = Annotated[T, "consume"]
_VALID_JSON = Annotated[Dict[str, Any], "json"]
JSON = Consume[_VALID_JSON]

reveal_type(JSON) # Type[T@JSON]
```

**VS Code extension or command-line**
VS Code: Pylance `v2022.8.20`

**Comment by erictraut** [comment]
The statement `Consume = Annotated[T, "consume"]` isn't a valid generic type alias. From the perspective of a static type checker, it's the equivalent of `Consume = T`. Pyright should probably generate an error in this case.

Can you provide more details about what you're trying to do here?

**Comment by NCPlayz** [comment]
Ah yeah, that does make sense. I was hoping to be able to narrow it down to `Annotated[Dict[str, Any], "json", "consume"]`.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by Gobot1234** [comment]
@erictraut The PEP seems to imply that code like this is valid https://peps.python.org/pep-0593/#aliases-concerns-over-verbosity, should this be brought up on https://github.com/python/typing?

**Comment by erictraut** [comment]
Thanks for pointing that out. I must have missed that section when I implemented the support for the PEP. Looks like it's explicitly called out as a supported case.

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is fixed in pyright 1.1.267, which I just published. It will also be included in a future version of pylance.
